bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/TypeChecker.java

```diff
@@ -691,12 +691,12 @@ private static boolean checkIsFunctionType(BType sourceType, BFunctionType targe
         }
 
         for (int i = 0; i < source.paramTypes.length; i++) {
-            if (!isSameType(source.paramTypes[i], targetType.paramTypes[i])) {
+            if (!checkIsType(targetType.paramTypes[i], source.paramTypes[i], new ArrayList<>())) {
                 return false;
             }
         }
 
-        return isSameType(source.retType, targetType.retType);
+        return checkIsType(source.retType, targetType.retType, new ArrayList<>());
```

Comment:
These changes would fix https://github.com/ballerina-platform/ballerina-lang/issues/13179 too right? Shall we verify and add tests for the cases mentioned in the issue too?

Reply:
Added with 4a207ce